### ISSUE TITLE
.NET 5

### ISSUE THREAD

**Issue by rbhanda** [dotnet/core#5545]
[Release Notes](https://github.com/dotnet/core/blob/master/release-notes/5.0/5.0.0/5.0.0.md)

Please report any issues you find with .NET 5, either responding to this issue, creating a new issue or creating a new issue in one of the following repos:

* [dotnet/aspnetcore](https://github.com/dotnet/aspnetcore/issues) - for ASP.NET Core questions
* [dotnet/efcore](https://github.com/dotnet/efcore/issues) - for Entity Framework Core questions
* [dotnet/sdk](https://github.com/dotnet/sdk/issues) - for CLI tools and questions
* [dotnet/runtime](https://github.com/dotnet/runtime/issues) - for runtime, API, or installer issues and questions
* [dotnet/winforms](https://github.com/dotnet/winforms/issues) - for WinForms issues
* [dotnet/wpf](https://github.com/dotnet/wpf/issues) - for WPF issues
* [nuget/home](https://github.com/nuget/home/issues) - for NuGet questions and issues

-----
Status of availability of SDK on Linux feeds: 

| Distro | 5.0.0 
|--|--|
| Ubuntu 16.04   | ✅  |
| Ubuntu 18.04| ✅  | 
| Ubuntu 20.04 | ✅  | 
| Ubuntu 20.10 | ✅  | 
| Centos 7   | ✅  | 
| Debian 9    | ✅  | 
| Debian 10   | ✅  | 
| Fedora 32   |  ✅ | 
| Fedora 33  |  ✅ | 
| OpenSUSE 15 | ✅  | 
| SLES 12     | ✅ | 
| SLES15   | ✅  | 

Note: This list refers to the Microsoft-provisioned feeds (packages.microsoft.com) and does not in any way represent direct availability in distros (eg RHEL, Fedora).

### Known Issues
If there are any issues with the November 2020 release we will track them here and check issues off as they're resolved. See the linked issues for details on progress and resolution details.

- [x] [Unable to install .NET 5 RC2, .NET Core 2.1, .NET Core 3.1 on Ubuntu 20.10](https://github.com/dotnet/core/issues/5352) - This is resolved for 5.0 and 3.1. See https://github.com/dotnet/core/issues/5017#issuecomment-729208437 for more details.

- [x] .NET 5 is missing in SLES15 https://github.com/dotnet/core/issues/5623

- [ ] Unable to build .NET 5 on some Linux distros (Fedora, CentOS, RHEL) https://github.com/dotnet/core/issues/5545#issuecomment-726803619

**Comment by bopin2020** [comment]
Anyone already had some same problems?  I tested .NET5 WPF on the linux system, but there was a big problem that I wanna why I used .NET5  to create a WPF project, as a result, I cat *csproj file <TargetFramework> contained "netcoreapp3.0". I need your help,please.
![image](https://user-images.githubusercontent.com/49940777/98786682-64ee3280-2439-11eb-872b-321c737a7ff2.png)

**Comment by RheaAyase** [comment]
These announcements are very misleading. .NET 5 is not "source-buildable" yet and therefor not available in many distributions. The checkmarks should be only partial.

And Fedora 30 and 31 shouldn't be in the list at all.

**Comment by bopin2020** [comment]
> These announcements are very misleading. .NET 5 is not "source-buildable" yet and therefor not available in many distributions. The checkmarks should be only partial.
> 
> And Fedora 30 and 31 shouldn't be in the list at all.

Yes! .NET5 cross platforms, but WPF not. "These announcements are very misleading" I am backing you absolutely!   I spent two days to figure it out if WPF could run on the linux system. I really need a UI project that can run on linux and finally I chose Avalonia.

**Comment by RheaAyase** [comment]
You can look into Avalonia if you haven't found one yet.

**Comment by RheaAyase** [comment]
Anyway my point is more about the way you get to install .NET 5 - it is not available in most Linux distributions like this checklist would lead you to believe. To take an example closest to me, Fedora. Fedora has dotnet 3.1 in its core repositories and Microsoft now refers to that as means of installing it via simply using Fedora's own package manager `dnf install dotnet` (or specifically `dnf install dotnet-sdk-3.1`) - however this is not the case with .NET 5 because it is not possible to build it from source yet. Microsoft prioritized producing it in binary form with the `source-build` being on the sidelines. As a result, the download page was also referring to .NET5 in Fedora repositories, while it's not there yet (this was patched today I believe) [2]

And on the 2nd paragraph - Fedora 30 reached end of life half a year ago and Fedora 31 is about to reach it next week. As such .NET 5 should not ship for either of those versions. It should not be mentioned in this list - it's not mentioned on [1] or [2] either.

And it's very similar case with RHEL - both 7 and 8 mentioned on [1] and [3] (And CentOS which is a clone of RHEL)

[1] https://docs.microsoft.com/en-us/dotnet/core/install/linux
[2] https://docs.microsoft.com/en-us/dotnet/core/install/linux-fedora
[3] https://docs.microsoft.com/en-us/dotnet/core/install/linux-rhel

---

TLDR: These checklists should not have checkmarks until the product is **actually** released for said platform.

**Comment by masters3d** [comment]
I did not see the below features in the release notes for dotnet 5. Are they moved donet 6?
> * Java interoperability will be available on all platforms.
> * Objective-C and Swift interoperability will be supported on multiple operating systems.

**Comment by mairaw** [comment]
@richlander @davidortinau @brunoborges can you comment on the questions from @masters3d?

**Comment by bopin2020** [comment]
> Anyway my point is more about the way you get to install .NET 5 - it is not available in most Linux distributions like this checklist would lead you to believe. To take an example closest to me, Fedora. Fedora has dotnet 3.1 in its core repositories and Microsoft now refers to that as means of installing it via simply using Fedora's own package manager `dnf install dotnet` (or specifically `dnf install dotnet-sdk-3.1`) - however this is not the case with .NET 5 because it is not possible to build it from source yet. Microsoft prioritized producing it in binary form with the `source-build` being on the sidelines. As a result, the download page was also referring to .NET5 in Fedora repositories, while it's not there yet (this was patched today I believe) [2]
> 
> And on the 2nd paragraph - Fedora 30 reached end of life half a year ago and Fedora 31 is about to reach it next week. As such .NET 5 should not ship for either of those versions. It should not be mentioned in this list - it's not mentioned on [1] or [2] either.
> 
> And it's very similar case with RHEL - both 7 and 8 mentioned on [1] and [3] (And CentOS which is a clone of RHEL)
> 
> [1] https://docs.microsoft.com/en-us/dotnet/core/install/linux
> [2] https://docs.microsoft.com/en-us/dotnet/core/install/linux-fedora
> [3] https://docs.microsoft.com/en-us/dotnet/core/install/linux-rhel
> 
> TLDR: These checklists should not have checkmarks until the product is **actually** released for said platform.

It help me a lot. Thank you very much!

**Comment by brunoborges** [comment]
> @richlander @davidortinau @brunoborges can you comment on the questions from @masters3d?

I'm not aware of what sort of Java interoperability exists in .NET.

**Comment by mairaw** [comment]
@brunoborges FYI this statement was included when we introduced .NET 5 
https://devblogs.microsoft.com/dotnet/introducing-net-5/
I can’t remember if we ever mentioned in subsequent posts what would happen to that feature. Was it just cut?

**Comment by HamidTheGeek** [comment]
I just update to `.NET 5` and I see this false positive for the result could be null.
![image](https://user-images.githubusercontent.com/33150399/99570615-1bb65980-29ca-11eb-9220-2abacf5bcabd.png)
As you can see the property is there and the result cannot be null.

**Comment by rbhanda** [comment]
Closing in favor of https://github.com/dotnet/core/issues/5719
